docs/dev/ui_description.rst

```diff
@@ -0,0 +1,264 @@
+
+            self.toolbar = self.components['custom-toolbar']
+
+            # This sets up the child-to-parent behavior
+            self.toolbar.observe(self.on_button_names_changed, names='button_names')
+
+            # Here we take advantage of the way trailets work
+
+        def on_button_names_changed(self, *args, **kwargs):
+            print("The list of button names has been changed.")
+
+This type of direct child-to-parent (i.e. the parent is responding to
+changes on the child) communication compliments the direct
+parent-to-child communication (i.e. the parent passing data to the
+child). However, this does not solve the application-level issue of
+multiple components, conceivably several levels deep, trying to interact
+with and pass data to each other. In this case, we decouple the widgets
+from each other and instead have them interact with a central,
+application-level communication hub through message objects.
+
+Global event handing
+~~~~~~~~~~~~~~~~~~~~
+
+Communication between widgets that do not have a direct reference to
+each other is handled using the ``Hub`` class of the glue-core package
+(a dependency of glue-jupyter). The hub implements the publish/subscribe
+paradigm wherein widgets subscribe to particular messages on the hub and
+are notified whenever those messages are published by any part of the
+UI. This system allows us to break hard dependencies between widgets in
+the UI that require passing references around and to develop widgets
+independently.
+
+The JDAViz package includes a base class that can be used for adding
+widgets that would need to communicate through global events. This is
+is the ``TemplateMixin`` class and allows passing glue session objects
+to widgets upon their instantiation. The glue session contains the ``Hub``
+object available to the application and it, along with other useful
+data objects, are easily accessible through the ``TemplateMixin``.
+
+Using the glue event framework is covered in great detail in the
+`glue documentation <http://docs.glueviz.org/en/stable/developer_guide/communication.html>`_.
+The code snippet gives an example of how an event listener may be
+implemented inside a widget::
+
+.. code:: python
+
+    from jdaviz.core.template_mixin import TemplateMixin
+    from glue.core.message import DataMessage
+
+    class TestWigdet(TemplateMixin):
+        text = Unicode("No messages...").tag(sync=True)
+        template = Unicode("""
+        <v-card>
+            <v-card-text>
+                <p>{{ text }}</p>
+            </v-card-text>
+        </v-card>
+        """).tag(sync=True)
+
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+
+            self.hub.subscribe(self, DataMessage,
+                               handler=self.on_data_message_received)
+
+        def on_data_message_received(self, message):
+            self.text = "Received data message!"
+
+Glue callback properties
```

Comment:
Just a section heading here with no text...Maybe add a "to be written...." if you are going defer writing this, just so we can merge and it doesn't look like it didn't build properly?